### ISSUE TITLE
Error on loading the page second time. 

### ISSUE THREAD

**Issue by a4armaan** [twinssbc/Ionic2-Calendar#527]
Ionic2-Calendar is not working after reopening the page 2nd time.

I am using it on the page and it loads properly. I am using the month view to display the calendar. I have a back button on my page to go back. After I reopen the calendar page second time I am getting bellow error. After that, all calendar functions stop working, Like loading events, slid next etc. It starts working after changing the Calendar mode. I have to change the Calendar mode from month to week or day, then change back to month then it starts working.

```
stack:"TypeError: Cannot read property 'dates' of undefined
    at MonthViewComponent.onDataLoaded (http://localhost/vendor.js:105786:440)
    at MonthViewComponent.ngOnChanges (http://localhost/vendor.js:105644:18)
    at MonthViewComponent.wrapOnChangesHook_inPreviousChangesStorage (http://localhost/vendor.js:56268:18)
    at callHook (http://localhost/vendor.js:39666:14)
    at callHooks (http://localhost/vendor.js:39632:17)
    at executeCheckHooks (http://localhost/vendor.js:39565:5)
    at refreshView (http://localhost/vendor.js:44280:21)
    at refreshDynamicEmbeddedViews (http://localhost/vendor.js:45339:21)
    at refreshView (http://localhost/vendor.js:44291:9)
    at refreshComponent (http://localhost/vendor.js:45403:9)"
__proto__:Error {constructor: , name: "TypeError", message: ""}
```
My ionic info :
```

Ionic:

   Ionic CLI                     : 6.4.1
   Ionic Framework               : @ionic/angular 5.0.7
   @angular-devkit/build-angular : 0.900.7
   @angular-devkit/schematics    : 9.1.1
   @angular/cli                  : 9.1.1
   @ionic/angular-toolkit        : 2.2.0

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.1.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.2.0, cordova-plugin-ionic-webview 4.1.3, (and 12 other plugins)

Utility:

   cordova-res : 0.11.0
   native-run  : 1.0.0

System:

   NodeJS : v12.16.1 (C:\Program Files\nodejs\node.exe)
   npm    : 6.13.4
   OS     : Windows 10
```

**Comment by twinssbc** [comment]
@a4arman According to the error stack, it seems the calendar fails to process some loaded data.
May I know what happened when the calendar is opened at the second time? Could you do some debugging to see if you load any different data when this error happens?

```
stack:"TypeError: Cannot read property 'dates' of undefined
    at MonthViewComponent.onDataLoaded (http://localhost/vendor.js:105786:440)
```

**Comment by alonsoavotz** [comment]
I have a similar problem.. when try for the first time, the calendar does not works.. but when refresh the page the calendar work fine
here is a video of the situation: https://youtu.be/hOncLfpqolU

plugin version> "ionic2-calendar": "^0.6.2",

my system info
Ionic:

   Ionic CLI                     : 5.2.8
   Ionic Framework               : @ionic/angular 5.0.7
   @angular-devkit/build-angular : 0.901.1
   @angular-devkit/schematics    : 9.1.1
   @angular/cli                  : 9.1.1
   @ionic/angular-toolkit        : 2.2.0

Cordova:

   Cordova CLI       : 8.0.0
   Cordova Platforms : android 7.0.0, ios 4.5.5
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 3.1.2, (and 17 other plugins)

Utility:

   cordova-res : 0.6.0 (update available: 0.12.1)
   native-run  : 0.2.9 (update available: 1.0.0)

System:

   ios-deploy : 1.9.4
   ios-sim    : 8.0.1
   NodeJS     : v10.15.3 (/usr/local/bin/node)
   npm        : 6.4.1
   OS         : macOS Catalina
   Xcode      : Xcode 11.2 Build version 11B52


Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.901.1
@angular-devkit/build-angular     0.901.1
@angular-devkit/build-optimizer   0.901.1
@angular-devkit/build-webpack     0.901.1
@angular-devkit/core              9.1.1
@angular-devkit/schematics        9.1.1
@angular/cli                      9.1.1
@ngtools/webpack                  9.1.1
@schematics/angular               9.1.1
@schematics/update                0.901.1
rxjs                              6.5.5
typescript                        3.8.3
webpack                           4.42.0

**Comment by twinssbc** [comment]
@alonsoavotz From your video, it seems related to the loaded data issue. Could you verify the difference between the data you load at the first time and second time?

**Comment by a4armaan** [comment]
> 
> 
> @a4arman According to the error stack, it seems the calendar fails to process some loaded data.
> May I know what happened when the calendar is opened at the second time? Could you do some debugging to see if you load any different data when this error happens?
> 
> ```
> stack:"TypeError: Cannot read property 'dates' of undefined
>     at MonthViewComponent.onDataLoaded (http://localhost/vendor.js:105786:440)
> ```

I am not loading any data. The mentioned error coming when page is open **second time**. Fist time it works fine. But after going back to previous page then coming back to calendar page i am getting mentioned error.

**Comment by a4armaan** [comment]
Now the interesting thing,
 Initially I had **Angular version 8.3.26**. After installing this (ionic2-calendar) package I have upgraded the Angular version 9. But mentioned error is coming. 
So I went through so many bugs comment here, then I took the old version of my repo, which has Angular version 8.3.26 and the installed **ionic2-calendar@0. 5.9**. 
Now with this combination its working perfect.

**Comment by a4armaan** [comment]
What is the solutions?

**Comment by pescador07** [comment]
Hello, 
I have the same problem. Any solution?
Thanks

**Comment by twinssbc** [comment]
This issue because a bug that was fixed in Ionic 5.1.0.
https://github.com/ionic-team/ionic/commit/3123a318b6755dbf8fde8476f81c37e6ffa9b70e#diff-e0874821af764a71e89442a158a223f8

You could upgrade the calendar to 0.6.4 version and @Ionic/angular to 5.1.0 to resolve it.
